### PR TITLE
Add host list filters

### PR DESCRIPTION
**Pull request by jtomasek**
Depends on ~~#1774~~, ~~#1697~~

### CODE REVIEW

File: frontend/packages/metal3-plugin/src/components/table-filters.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { getSimpleHostStatus } from '../utils/host-status';
+import { HOST_STATUS_TITLES } from '../constants';
+
+type FilterItem = {
+  id: string;
+  title: string;
+};
+
+type Filter = {
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/table-filters.ts]: @vojtechszocs Those types are common across all filters (IIUC), should I move them into `@console/internal/components/factory/table-filters` to make them widely available?

Comment (mareklibra) [reply frontend/packages/metal3-plugin/src/components/table-filters.ts]: Within #1689, I am moving them under `@console/shared`

Comment (jtomasek) [reply frontend/packages/metal3-plugin/src/components/table-filters.ts]: Rebased on that change

File: frontend/packages/metal3-plugin/src/components/table-filters.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import {
+  HOST_REGISTERING_STATES,
+  HOST_PROVISIONING_STATES,
+  HOST_STATUS_READY,
+  HOST_ERROR_STATES,
+  HOST_STATUS_TITLES,
+  HOST_STATUS_PROVISIONED,
+} from '../constants';
+
+const hostStatesToFilterMap = {
```
Comment (spadgett) [inline frontend/packages/metal3-plugin/src/components/table-filters.ts]: I suggest `Object.freeze` here

File: frontend/packages/metal3-plugin/src/components/table-filters.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+const getHostFilterStatus = (host: K8sResourceKind): string => {
+  const status = getSimpleHostStatus(host);
+  return _.findKey(hostStatesToFilterMap, ({ states }) => states.includes(status));
+};
+
+export const hostStatusFilter: Filter = {
+  type: 'host-status',
+  selected: Object.keys(hostStatesToFilterMap),
+  reducer: getHostFilterStatus,
+  items: Object.keys(hostStatesToFilterMap).map((state) => ({
```
Comment (spadgett) [inline frontend/packages/metal3-plugin/src/components/table-filters.ts]: Suggest `_.map`

```suggestion
  items: _.map(hostStatesToFilterMap, ({title}, id) => ({ id, title })),
```

File: frontend/packages/metal3-plugin/src/components/table-filters.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    id: state,
+    title: hostStatesToFilterMap[state].title,
+  })),
+  filter: (statuses, host) => {
```
Comment (spadgett) [inline frontend/packages/metal3-plugin/src/components/table-filters.ts]: Add types when possible

File: frontend/packages/metal3-plugin/src/components/table-filters.ts (hunk 2 above)
Comment (jtomasek) [reply frontend/packages/metal3-plugin/src/components/table-filters.ts]: Done

File: frontend/packages/metal3-plugin/src/components/table-filters.ts (hunk 3 above)
Comment (jtomasek) [reply frontend/packages/metal3-plugin/src/components/table-filters.ts]: Done

File: frontend/packages/metal3-plugin/src/components/table-filters.ts (hunk 4 above)
Comment (jtomasek) [reply frontend/packages/metal3-plugin/src/components/table-filters.ts]: Done

**Pull request merged**